apps/explorer/src/ui/CopyToClipboard.tsx

```diff
@@ -28,31 +34,65 @@ const iconStyles = cva(['cursor-pointer hover:text-steel-dark'], {
 
 export type IconStylesProps = VariantProps<typeof iconStyles>;
 
-export interface CopyToClipboardProps extends IconStylesProps {
+export interface CopyToClipboardProps extends Omit<IconStylesProps, 'success'> {
 	copyText: string;
 	onSuccessMessage?: string;
 }
 
-const ICON_SIZES = {
+const COPY_ICON_SIZES = {
 	sm: Copy12,
 	md: Copy16,
 	lg: Copy24,
 };
 
+const CHECK_ICON_SIZES = {
+	sm: Check12,
+	md: CheckStroke16,
+	lg: CheckStroke24,
+};
+
+const TIMEOUT_TIMER = 2000;
+
 export function CopyToClipboard({
 	copyText,
 	color,
 	size = 'md',
 	onSuccessMessage = 'Copied!',
 }: CopyToClipboardProps) {
+	const [showCopyIcon, setShowCopyIcon] = useState(true);
```

Comment:
I think we should invert this, i.e. 

```
const [copied, setCopied] = useState(false)
```

and then

```
useEffect(() => {
  if (copied) {
    const timeout = setTimeout(() => { 
      setCopied(false)
     })
  }
}, [])
```

Reply:
updated